### ISSUE TITLE
Embed gist on gatsby-mdx page

### ISSUE THREAD

**Issue by tuhlmann** [weirdpattern/gatsby-remark-embed-gist#12]
I'm trying to use your plugin on a gatsby-mdx page. 

Here's my gatsby-config snippet:
```
    {
      resolve: "gatsby-mdx",
      options: {
        defaultLayouts: {
          default: here("./src/templates/markdown-page.js"),
        },
        extensions: [".mdx", ".md", ".markdown"],
        gatsbyRemarkPlugins: [
          { resolve: "gatsby-remark-copy-linked-files" },
          {
            resolve: "gatsby-remark-images",
            options: {
              backgroundColor: "#fafafa",
              maxWidth: 1035,
              sizeByPixelDensity: true,
            },
          },
          { resolve: require.resolve("./plugins/remark-embedder") },
          {
            resolve: "gatsby-remark-embed-gist",
            options: {
              username: "tuhlmann",

              // a flag indicating whether the github default gist css should be included or not
              // default: true
              includeDefaultCss: true
            }
          },
        ],
      },
    },
```

The gist I'm trying to include: `gist:tuhlmann/d9f1e3237eb8f692eb71`

Embedding a gist throws the following error:

```
error Plugin gatsby-mdx returned an error


  SyntaxError: unknown: Unexpected token (291:138)
    289 |       <tr>
    290 |         <td id="file-core-clj-L54"   class="blob-num js-line-number" data-line-number=  9m"54"></td>
  > 291 |         <td id="file-core-  clj-LC54" class="blob-code blob-code-inner js-file-line">  <span class="pl-s"><span  9m class="pl-pds">&quot;</  [39mspan>{</span></  3mtd>
        |
                                              ^
    292 |       </tr>
    293 |       <tr>
    294 |         <td id="file-core-clj-L55"   class="blob-num js-line-number" data-line-number=  9m"55"></td>
```

Is there a way to work around this problem?

Thanks for providing this plugin!

Torsten.

**Comment by weirdpattern** [comment]
Hi @tuhlmann, it looks like one or more plugins are acting on top of the output generated by gatsby-remark-embed-gist... Is that the complete configuration you are using or is there something else after the gatsby-remark-embed-gist plugin?

Try playing with the order of the plugins (see the notes section in the documentation).

LMK

**Comment by tuhlmann** [comment]
I put the gist plugin at the top of the mdx plugins, but that didn't help. For reference, here's my complete `gatsby-config.js` the way it was initially (before rearranging anything): https://gist.github.com/tuhlmann/b576df42bdf1585560b22bb5fc048fb9

Thanks,
Torsten.

**Comment by weirdpattern** [comment]
Hey @tuhlmann, I reviewed this yesterday.
As far as I can tell the transformation does happen, but then MDX does something with the generated HTML that causes the directives to be rendered twice thus the unmatched brackets...

I also tried updating the plug-in to use mutateSource instead of the visitor pattern used by default, but it seems mdx doesn't support this as per ChristopherBiscardi/gatsby-mdx#118

I'm gonna keep trying tonight...

**Comment by tuhlmann** [comment]
Thanks for looking into it!

**Comment by weirdpattern** [comment]
Alright, I just merged a commit that fixes this issue, but this only solves half of your problem.

The plugin contains two files.
- The index file defines the remark transformation.
- The gatsby-ssr file imports the gist stylesheet into your html.

Unfortunately, gatsby-mdx only executes the former, so the gist renders without styles which looks weird  and probably defeats the purpose of the plug-in... but no worries, I have two solutions for you...

1. Manually importing the gist stylesheet

```css
@import url("https://github.githubassets.com/assets/gist-embed-b3b573358bfc66d89e1e95dbf8319c09.css");
``` 

2. Add "gatsby-remark-embed-gist/gatsby-ssr` to your list of plugins outside the gatsby-mdx configuration

```js
plugins: [
   ...
   {
      resolve: "gatsby-mdx",
      options: {
        defaultLayouts: {
          default: path.resolve('./src/components/layout.js')
        },
        extensions: [".mdx", ".md", ".markdown"],
        gatsbyRemarkPlugins: [
          { resolve: "gatsby-remark-copy-linked-files" },
          {
            resolve: "gatsby-remark-images",
            options: {
              backgroundColor: "#fafafa",
              maxWidth: 1035,
              sizeByPixelDensity: true,
            },
          },
          {
            resolve: "gatsby-remark-embed-gist",
            options: {
              username: "xxxx"
            }
          }
        ],
      },
    },
    ...
    "gatsby-remark-embed-gist/gatsby-ssr",
    ...
]
```

I tried and both worked for me...

**Comment by weirdpattern** [comment]
Closed by #13

**Comment by weirdpattern** [comment]
v.1.1.7 has been published
